src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,96 @@
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList = ({
+  todos,
+  filter,
+  search,
+  todoModalId,
+  onModalButtonClick,
+}: Props) => {
+  const filteredTodos =
+    todos?.filter(todo => {
+      if (filter === 'active') {
+        return !todo.completed;
+      }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      if (filter === 'completed') {
+        return todo.completed;
+      }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      return true;
+    }) || null;
+
+  if (!filteredTodos) {
+    throw new Error('Todos is missing');
+  }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  const searchTodo = filteredTodos.filter(todo =>
+    todo.title.toLowerCase().includes(search.toLowerCase()),
+  );
+
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {searchTodo?.map(todo => {
+          return (
+            <tr key={todo.id} data-cy="todo" className="">
```

Comment:
Keys should not be generated on render. This is because if the keys change then the component will unmount and remount. This can lead to issues like losing local state and triggering unnecessary effects. Instead, consider using unique and stable keys. For example, if each todo has a unique id, you can use that as the key.

Reply:
I do xD